smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceIdentifierBindingValidator.java

```diff
@@ -227,13 +228,24 @@ private Stream<ValidationEvent> validateInstanceBindings(
     }
 
     private boolean hasAllIdentifiersBound(
+            Model model,
             ResourceShape resource,
             OperationShape operation,
             IdentifierBindingIndex bindingIndex
     ) {
-        return bindingIndex.getOperationInputBindings(resource, operation)
-                .keySet()
-                .containsAll(resource.getIdentifiers().keySet());
+        StructureShape inputShape = model.expectShape(operation.getInputShape(), StructureShape.class);
+        Map<String, String> bindings = bindingIndex.getOperationInputBindings(resource, operation);
+        for (Map.Entry<String, ShapeId> identifier : resource.getIdentifiers().entrySet()) {
+            if (!bindings.containsKey(identifier.getKey())) {
```

Comment:
I think in-lining these if's statements (into return) might be a bit more readable

Reply:
```
if (!bindings.containsKey(identifier.getKey())
                    || !inputShape.getMember(bindings.get(identifier.getKey()))
                            .get()
                            .getTarget()
                            .equals(identifier.getValue())) {
                return false;
            }
```

This didn't look as readable to me as the split.